packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -295,3 +303,84 @@ test('should announce change to network', async () => {
   await ceramic.repository.stateManager.applyCommit(streamState1.id, updateRec, { anchor: false, publish: true });
   expect(publishTip).toHaveBeenCalledWith(stream1.id, stream1.tip);
 });
+
+describe('sync', () => {
+  let originalCeramic: Ceramic;
+
+  beforeEach(() => {
+    originalCeramic = ceramic;
+  });
+
+  afterEach(() => {
+    ceramic = originalCeramic;
+  });
+
+  const FAKE_STREAM_ID = StreamID.fromString('kjzl6cwe1jw147dvq16zluojmraqvwdmbh61dx9e0c59i344lcrsgqfohexp60s');
+  function digest(input: string) {
+    return uint8arrays.toString(sha256.hash(uint8arrays.fromString(input)), 'base16');
+  }
+  function hash(data: string): CID {
+    return new CID(1, 'sha2-256', Buffer.from('1220' + digest(data), 'hex'));
+  }
+
+  function responseTips(amount: number) {
+    const times = Array.from({ length: amount }).map((_, index) => index);
+    return times.map((n) => hash(n.toString()));
+  }
+
+  test('handle first received', async () => {
+    const stateManager = ceramic.repository.stateManager;
+    const response = responseTips(1);
+    ceramic.dispatcher.messageBus.queryNetwork = () => from(response);
+    const fakeHandleTip = jest.fn(() => Promise.resolve());
+    (stateManager as any)._handleTip = fakeHandleTip;
+    const state$ = ({ id: FAKE_STREAM_ID } as unknown) as RunningState;
+    await stateManager.sync(state$, 1000);
+    expect(fakeHandleTip).toHaveBeenCalledWith(state$, response[0]);
+  });
+  test('handle all received', async () => {
+    const stateManager = ceramic.repository.stateManager;
+    const amount = 10;
+    const response = responseTips(amount);
+    ceramic.dispatcher.messageBus.queryNetwork = () => from(response);
+    const fakeHandleTip = jest.fn(() => Promise.resolve());
+    (stateManager as any)._handleTip = fakeHandleTip;
+    const state$ = ({ id: FAKE_STREAM_ID } as unknown) as RunningState;
+    await stateManager.sync(state$, 1000);
+    response.forEach((r) => {
+      expect(fakeHandleTip).toHaveBeenCalledWith(state$, r);
+    });
+  });
+  test('not handle delayed', async () => {
+    const stateManager = ceramic.repository.stateManager;
+    const amount = 10;
+    const response = responseTips(amount);
+    ceramic.dispatcher.messageBus.queryNetwork = () =>
+      from(response).pipe(
+        concatMap(async (value, index) => {
+          await new Promise((resolve) => setTimeout(resolve, index * MAX_RESPONSE_INTERVAL * 0.3));
```

Comment:
could you maybe add a comment to the start of this test explaining what it does and how the timing math works out?  It's not obvious to me reading this why we're multiplying this by 0.3